staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -155,6 +155,9 @@ type Config struct {
 	// If specified, long running requests such as watch will be allocated a random timeout between this value, and
 	// twice this value.  Note that it is up to the request handlers to ignore or honor this timeout. In seconds.
 	MinRequestTimeout int
+	// MinimalShutdownDuration allows to block shutdown for some time, e.g. until endpoints pointing to this API server
+	// have converged on all node. During this time, the API server keeps serving.
+	MinimalShutdownDuration time.Duration
```

Comment:
Can we come up with a clearer name? I'm not sure what "minimal" is adding.

LameDuckDuration? PostShutdownServingDuration? ServeDuringGracePeriod?

Do we need two flags? One to set the period during which it accepts new (short running) requests, and one to set the period during which it rejects new requests but continues to serve old requests? e.g. PostShutdownAcceptRequestsDuration / PostShutdownServeExistingRequestsDuration.

It is a little weird to me that that this is a duration, rather than just waiting for an external system to complete the shutdown process after it confirms that we are no longer in the relevant load balancers.

Reply:
Flags are kind of forever so I'd prefer to get it right if possible :/

Reply:
not sure we need the second duration. The only use-case I can come up with is that `/healthz` keeps returning 200. But as we are shutting down anyway, the kubelet does not care anymore about `/healthz`.

About an external system: this depends a lot on the environment, whether the load balancers are purely health check based and there is no further communication between the cluster and the LB. That setup has the advantage that it is much simpler, available with every LB, and there is no risk for cyclic dependencies, especially helpful for control plane components.

We might want to support both: pure health check based + some external control.

Reply:
Renamed to `--shutdown-delay-period`.